### PR TITLE
Support for Freestyle Precision/Optium Neo (UPLOAD-218)

### PR DESCRIPTION
**Pull request by gniezen**
Addresses:

- [UPLOAD-218]

### CODE REVIEW

File: lib/drivers/abbott/freeStyleLibreProtocol.js [hunk 1]
```diff
@@ -610,13 +610,13 @@ export default class FreeStyleLibreProtocol {
...
 
-      dbRecords = dbRecords.split('\r\n');
+      dbRecords = dbRecords.split('\r\n').filter(Boolean);
 
       if (dbRecordCount !== dbRecords.length) {
         return errorCallback(new Error(`Invalid database record count: ${dbRecordCount} != ${dbRecords.length}`), data);
       }
 
-      return successCallback(null, dbRecords);
+      return successCallback(dbRecords);
```
Comment (pazaan) [inline lib/drivers/abbott/freeStyleLibreProtocol.js]: This signature is different. Does this fix a bug? Was data previously not being returned?

File: lib/drivers/abbott/abbottFreeStyleNeo.js [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+      },
+      (err) => {
+        debug('requestTextResponse Error:', err);
+        reject(err);
+      });
+    });
+  }
+
+  async getSerialNumber() {
+    const serialNumber = await this.commandResponse('$serlnum?');
```
Comment (pazaan) [inline lib/drivers/abbott/abbottFreeStyleNeo.js]:
```suggestion
    return this.commandResponse('$serlnum?');
```

And delete the next line.

File: lib/drivers/abbott/abbottFreeStyleNeo.js [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+        );
+      }).catch((error) => {
+        debug('Error in getConfigInfo: ', error);
+        return cb(error, null);
+      });
+    },
+
+    fetchData(progress, data, cb) {
+      debug('in fetchData', data);
+      (async () => {
```
Comment (pazaan) [inline lib/drivers/abbott/abbottFreeStyleNeo.js]: If you wanted to, you can remove the "then/catch" Promise handling with:
```
(async () => {
  try {
    data.records = await driver.getRecords();
    progress(100);
    return cb(null, data);
  } catch (error) {
    return cb(error, null);
  }
})()
```

File: lib/drivers/abbott/freeStyleLibreProtocol.js (hunk 1 above)
Comment (gniezen) [reply lib/drivers/abbott/freeStyleLibreProtocol.js]: `getDbRecords` is not actually being used in the Libre driver, as the Libre uses the binary protocol and not the text protocol. This is documented in the comments on lines 623 and 630, but not in the function itself. To be sure, I'm going to add a smoke test of the Libre reader to the test strategy.

File: lib/drivers/abbott/abbottFreeStyleNeo.js (hunk 3 above)
Comment (gniezen) [reply lib/drivers/abbott/abbottFreeStyleNeo.js]: Yeah, I like that idea.

File: lib/drivers/abbott/abbottFreeStyleNeo.js (hunk 2 above)
Comment (gniezen) [reply lib/drivers/abbott/abbottFreeStyleNeo.js]: :+1:

**Pull request merged**